### PR TITLE
Arm64/Sve: Add SignExtendWidening* and ZeroExtendWidening* math APIs

### PR DESCRIPTION
**Pull request by kunalspathak**
- SignExtendWideningLower
- SignExtendWideningUpper
- ZeroExtendWideningLower
- ZeroExtendWideningUpper

New tests added are passing: https://gist.github.com/kunalspathak/7d2dc1578786410a45d39f54754af4a1

Contributes to https://github.com/dotnet/runtime/issues/99957

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/arm64-contrib

**Review by TIHan — APPROVED** [review]
LGTM, thanks!

**Comment by kunalspathak** [comment]
/ba-g Failure is https://github.com/dotnet/runtime/issues/101721

File: src/coreclr/jit/emitarm64sve.cpp [hunk 1]
```diff
@@ -2185,7 +2185,7 @@ void emitter::emitInsSve_R_R(instruction     ins,
         case INS_sve_uunpkhi:
         case INS_sve_uunpklo:
             assert(insScalableOptsNone(sopt));
-            assert(insOptsScalableAtLeastHalf(opt));
```
Comment (a74nh) [inline src/coreclr/jit/emitarm64sve.cpp:2188]: Does doing this break the unit tests? I would hope there are some H,S,D tests that need switching to B,H,S

**Pull request merged**